### PR TITLE
use stable stringify for better objects diff

### PR DESCRIPTION
**Pull request by iamdoron**
afterstable

instead of:
beforestable

### CODE REVIEW

**Comment by Marsup** [comment]
This won't work with a circular object.

**Comment by iamdoron** [comment]
what do you mean by 'it won't work'? It should act the same as before

**Comment by Marsup** [comment]
My bad, I thought json-stringify-safe was more clever than it really is.

**Comment by iamdoron** [comment]
Yes, it just outputs `"[Circular]"`.

File: lib/reporters/console.js [hunk 1]
```diff
@@ -13,6 +14,12 @@ const internals = {
     width: 50
 };
 
+internals.stringify = (obj, serializer, indent, decycler) => {
+
+    return StableStringify(
```
Comment (Marsup) [inline lib/reporters/console.js]: Can't we just do `StableStringify(obj, { space: indent, replacer: StringifySafe.getSerialize(serializer) })` ?

Comment (iamdoron) [inline lib/reporters/console.js]: if you do something like
```javascript
internals.stringify = (obj, serializer, indent) => {

    return StableStringify(
      JSON.parse(StringifySafe(obj)),
      { space: indent, replacer: serializer });
};
```

The test 'console reports with undefined values fails'

After the first stringify we lose some information about the original object:
```javascript
JSON.stringify({a: undefined}) === '{}'
JSON.stringify({a: Infinity}) === '{"a":null}'
```

Comment (iamdoron) [inline lib/reporters/console.js]: yes, we can 👍 
I changed it

**Comment by arb** [comment]
I would maybe recommend "fast-safe-stringify" as an alternative since I think it's the fastest one out that that doesn't crash on circular references.

**Comment by iamdoron** [comment]
@arb - using @Marsup  suggestion, it is better to use `StringifySafe.getSerialize()`, as it can be composed with `StableStringify` seamlessly. There's no such interface in `fast-safe-stringify`

**Comment by geek** [comment]
@iamdoron well done, this will be a nice improvement for our console reporter.

**Pull request merged**